src/components/huskelapp/HuskelappModal.tsx

```diff
@@ -33,88 +54,145 @@ const ModalContent = styled(Modal.Body)`
+      case Oppfolgingsgrunn.VURDER_TILTAK_BEHOV:
+        return texts.oppfolgingsgrunn.vurderTiltakBehov;
+      case Oppfolgingsgrunn.ANNEN_OPPFOLGNING:
+        return texts.oppfolgingsgrunn.annenOppfolgning;
+    }
   };
 
-  const handleOppdaterHuskelapp = () => {
-    const huskelappDto: HuskelappRequestDTO = { tekst: tekst };
-    oppdaterHuskelapp.mutate(huskelappDto, {
-      onSuccess: () => toggleOpen(false),
-    });
+  const handleOnSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    if (oppfolgingsgrunn !== undefined) {
+      const huskelappDto: HuskelappRequestDTO = {
+        oppfolgingsgrunn: oppfolgingsgrunn,
+      };
+      oppdaterHuskelapp.mutate(huskelappDto, {
+        onSuccess: () => toggleOpen(false),
+      });
+    } else {
+      setIsFormError(true);
+      return undefined;
+    }
+  };
+
+  const handleOppfolgingsgrunnChange = (oppfolgingsgrunn: Oppfolgingsgrunn) => {
+    setOppfolgingsgrunn(oppfolgingsgrunn);
+    setIsFormError(false);
   };
 
   const handleRemoveHuskelapp = (uuid: string) => {
     removeHuskelapp.mutate(uuid, {
       onSuccess: () => toggleOpen(false),
     });
   };
+  const hasHuskelapp = !!huskelapp;
 
   return (
-    <Modal
-      closeOnBackdropClick
-      className="px-6 py-4 w-full max-w-[50rem]"
-      aria-label={"huskelapp"}
-      open={isOpen}
-      onClose={() => toggleOpen(false)}
-      header={{ heading: texts.header }}
-    >
-      <ModalContent>
-        {isLoading && <HuskelappSkeleton />}
-        {isSuccess && (
-          <Textarea
-            label={texts.textAreaLabel}
-            hideLabel
-            defaultValue={huskelapp?.tekst}
-            onChange={oppdaterTekst}
-          />
-        )}
-      </ModalContent>
-      <Modal.Footer>
-        <Button
-          variant="primary"
-          onClick={handleOppdaterHuskelapp}
-          loading={oppdaterHuskelapp.isLoading}
-          disabled={isLoading}
-        >
-          {texts.save}
-        </Button>
-        <Button variant="secondary" onClick={() => toggleOpen(false)}>
-          {texts.close}
-        </Button>
-        {huskelapp && (
-          <Tooltip content={texts.removeTooltip}>
+    <form onSubmit={handleOnSubmit}>
+      <Modal
+        closeOnBackdropClick
+        className="px-6 py-4 w-full max-w-[50rem]"
+        aria-label={"huskelapp"}
+        open={isOpen}
+        onClose={() => toggleOpen(false)}
+        header={{ heading: texts.header }}
+      >
+        <ModalContent>
+          {isLoading && <HuskelappSkeleton />}
+          {hasHuskelapp ? (
+            <BodyShort>{oppfolgingsgrunnToText()}</BodyShort>
+          ) : (
+            <RadioGroup
+              legend={texts.oppfolgingsgrunn.label}
+              onChange={handleOppfolgingsgrunnChange}
+              size="small"
+              error={isFormError && texts.missingOppfolgingsgrunn}
+            >
+              <Radio value={Oppfolgingsgrunn.AVVENT_DIALOGMOTE}>
```

Comment:
Hadde det vært finere med en map her, i stedet for å manuelt lage en radio for hver grunn? Evt. skille det ut gruppen i en egen komponent? Eller kanskje det bare hadde blitt vanskelig å lese 🤔

Reply:
Liker forslaget! Iterere over et objekt med oppfolgingsgrunner som key og tekst som value kunne vært nice :+1: